### PR TITLE
[sdk/python]: improve codegen to significantly reduce size of generated models

### PR DESCRIPTION
**Pull request by Jaguar0625**
problem: generated code is very duplicative
    solution: add concept of base class to generator to reduce duplication

### CODE REVIEW

File: sdk/python/symbolchain/sc/__init__.py [hunk 1]
```diff
@@ -635,12 +637,11 @@ def size(self) -> int:
 		return size
 
 	@classmethod
-	def deserialize(cls, payload: bytes | bytearray | memoryview) -> Transaction:
-		buffer = memoryview(payload)
+	def _deserialize(cls, buffer: memoryview, instance) -> (int, int):
+		size_ = len(buffer)
 		size_ = int.from_bytes(buffer[:4], byteorder='little', signed=False)
 		buffer = buffer[4:]
 		buffer = buffer[:size_ - 4]
```
Comment (Wayonb) [inline sdk/python/symbolchain/sc/__init__.py:644]: Can this be rewritten as ``buffer = buffer[4:size_ - 4]``

File: sdk/python/symbolchain/sc/__init__.py [hunk 2]
```diff
@@ -662,18 +663,22 @@ def deserialize(cls, payload: bytes | bytearray | memoryview) -> Transaction:
...
+		# pylint: disable=protected-access
 		instance._signature = signature
 		instance._signer_public_key = signer_public_key
 		instance._version = version
 		instance._network = network
 		instance._type_ = type_
 		instance._fee = fee
 		instance._deadline = deadline
-		return instance
+		return (size_ - len(buffer), size_)
```
Comment (Wayonb) [inline sdk/python/symbolchain/sc/__init__.py:674]: Should ``len(buffer)`` be zero at end? 🤔

**Review by Wayonb — CHANGES_REQUESTED** [review]

File: sdk/python/symbolchain/sc/__init__.py [hunk 3]
```diff
@@ -635,12 +637,11 @@ def size(self) -> int:
 		return size
 
 	@classmethod
-	def deserialize(cls, payload: bytes | bytearray | memoryview) -> Transaction:
-		buffer = memoryview(payload)
+	def _deserialize(cls, buffer: memoryview, instance) -> (int, int):
+		size_ = len(buffer)
```
[the remark this answers is not in the document]
Comment (Jaguar0625) [reply sdk/python/symbolchain/sc/__init__.py:641]: this was added for NEM, where structs don't have intrinsic side fields. i will add a better filter.

File: sdk/python/symbolchain/sc/__init__.py (hunk 1 above)
Comment (Jaguar0625) [reply sdk/python/symbolchain/sc/__init__.py:644]: i can tweak the generator a bit 👍

File: sdk/python/symbolchain/sc/__init__.py (hunk 2 above)
Comment (Jaguar0625) [reply sdk/python/symbolchain/sc/__init__.py:674]: no - buffer is entire *remaining* buffer. so, we need to effectively shift (and mask) it like in the JS implementation:
```
		view.shiftRight(4);
		view.shrink(size - 4);
```
remember, this is base class (Transaction) and size includes both base + derived fields. at this point, buffer should contain only derived part. `size_ - len(buffer) ` let's us skip to derived part. `size_` masks off the next entity(ies), if present [important for aggregates, which have cosignatures expanding to fill remaining space].

**Review by Wayonb — APPROVED** [review]
👍🏾

**Pull request merged**